src/rust/runtime/memory/demibuffer.rs

```diff
@@ -508,6 +523,63 @@ impl DemiBuffer {
         self.as_metadata().data_len as usize
     }
 
+    /// Returns the amount of headroom available in the packet.
+    pub fn headroom(&self) -> u16 {
+        self.as_metadata().data_off
+    }
+
+    /// Returns a flag indicating whether this DemiBuffer is direct, i.e., that this instance is
+    /// not a clone of some other DemiBuffer.
+    pub fn is_direct(&self) -> bool {
+        self.as_metadata().ol_flags & METADATA_F_INDIRECT == 0
+    }
+
+    /// Get the direct DemiBuffer from an indirect DemiBuffer.
+    pub fn into_direct(self) -> DemiBuffer {
```

Comment:
Can this be refactored into smaller fn()?

Reply:
done